group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -8195,6 +8196,7 @@ public void createGroupTombstoneRecords(
         List<CoordinatorRecord> records
     ) {
         group.createGroupTombstoneRecords(records);
+        timer.cancel(streamsInitialRebalanceKey(group.groupId()));
```

Comment:
Other timers are cancelled using a dedicated cancel function like this:

```
    /**
     * Cancels the sync timeout of the member.
     *
     * @param groupId       The group id.
     * @param memberId      The member id.
     */
    private void cancelConsumerGroupSyncTimeout(
        String groupId,
        String memberId
    ) {
        timer.cancel(consumerGroupSyncKey(groupId, memberId));
    }
```

Best to follow that pattern.